src/generators.jl

```diff
@@ -802,18 +802,159 @@ end
 # TODO Add chung-lu for general floating point weights
 # via union of ER graphs add 
 
+
 """
-`forest_fire_graph`
-===================
+`forest fire graph`
+=====================
 
-Create an instance of a forest fire graph. The forest fire
-model consists of ...
+Generate a forest fire network with burn probability `burn_p` of `target_size`
+from a given undirected `seed_network`. The final network is built one vertex at
+a time which gain neighbors by "burning" the edges along a random walk starting
+from a random pre-existing node. 
 
-Usage
+Input
 -----
-`A = forest_fire_graph(c, k, p)` returns a forest-fire sample. 
-`A = forest_fire_graph(A0, k, p)` 
-"""
+- 'A::Union{MatrixNetwork,SparseMatrixCSC}': An undirected seed Network. 
+- `target_sizes::Int': The final number of vertices of the target graph.
+- `burn_p::Float64': The probability of success when drawing the number of 
+   neighbors to burn from a geometric distribution. 
+- `seed::Union{Nothing,UInt}' - An optional seed for the network.
+- `random_node'::Int -> Int  - The function to randomly select parents. 
+Reference: 
+
+    (section 4.2.1) Graph Evolution: Densification and Shrinking Diameters
+    Jure Leskovec, Jon Kleinberg, Christos Faloutsos
+    https://arxiv.org/pdf/physics/0603229.pdf
+
+Functions
+---------
+* `forest_fire_graph(n)` Generate an n node graph from a 10 node clique with a .4 
+   burn probability. 
+* `forest_fire_graph(G,n,p)` Generate an n node graph from the seed graph G, with a 
+   burn probability of p. 
+* `forest_fire_graph(::Type{S},G,n,p)` Generate an n node graph from the seed 
+   graph G, with a probability of p for vertex labels of type S. 
+
+Output
+------
+- `A` - the newly generated network of the same type as `seed_network`.
+- 'parents::Array{Int,1}': The list of vertices selected to start the random
+   walks from. Vertices who are their own parents are from the seed_network.
+""" 
+function forest_fire_graph(target_size::Int,m::Int=10, 
+    burn_p::Float64=.4; kwargs...)
+    clique = sprand(m,m,1.0)
+    clique[1:(m+1):m^2] .= 0
+    dropzeros!(clique)
+    return forest_fire_graph(eltype(clique.rowval),clique,target_size,burn_p;kwargs...)
+end 
+
+forest_fire_graph(seed_network::Union{MatrixNetwork,SparseMatrixCSC},target_size::Int, burn_p::Float64;kwargs...) = 
+                forest_fire_graph(eltype(seed_network.rowval),seed_network,target_size, burn_p; kwargs...)
+
+function forest_fire_graph(::Type{S},seed_network::Union{MatrixNetwork,SparseMatrixCSC}, 
+                           target_size::Int, burn_p::Float64;
+                           seed::Union{Nothing,UInt}=nothing,
+                           random_node=x->rand(1:x)) where S
+
+    if !(seed === nothing)
```

Comment:
Can you replace this with specifying the RNG where Random.GlobalRNG is the default? Then they can give StableRNG(seed) if they want reliable seeds...

Reply:
GIven that we let the user pass in the `random_node` function as well, how should I adapt it to work with an rng variable? 

Seems like `random_node=(x,rng)->rand(rng,1:x))` would force the user to think about a seed if they want to adjust the parent's probability of being selected.

Reply:
The convention is for rand functions in Julia to take in an optional random number generator to avoid using the global one. Users should have to think about how to make their random results reproducible if they want them :), but it should be easy enough ... The random node stuff is just a placeholder right now, let's not worry about it.